### PR TITLE
Make PushQueue test less flaky

### PR DESCRIPTION
**Pull request by howardjohn**
The previous test depended on ordering of goroutines which is not
reliable. It would consistently fail when running repeatedly. This
change refactors the test to provide more coverage while not relying on
any undefined ordering.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/pushqueue_test.go [hunk 1]
```diff
@@ -182,35 +183,48 @@ func TestProxyQueue(t *testing.T) {
...
-		wg.Add(1)
+		}
 		go func() {
-			expect := 0
-			for {
-				ExpectDequeue(t, p, proxies[expect])
-				expect++
-				if expect == len(proxies) {
-					wg.Done()
-					break
```
Comment (mandarjog) [inline pilot/pkg/proxy/envoy/v2/pushqueue_test.go]: This is just testing that there is nothing left in the queue after at most 10s.
Is there anything else this should be testing ?

Comment (howardjohn) [reply pilot/pkg/proxy/envoy/v2/pushqueue_test.go]: it's checking `expected` is empty meaning we have received all of the pushes we expect - first we populate it with what we expect and remove them as we see them

**Pull request merged**